src/installer/corehost/corehost.cpp

```diff
@@ -213,11 +213,24 @@ int exe_start(const int argc, const pal::char_t* argv[])
         trace::info(_X("App path: [%s]"), app_path.c_str());
 
         hostfxr_set_error_writer_fn set_error_writer_fn = reinterpret_cast<hostfxr_set_error_writer_fn>(pal::get_symbol(fxr, "hostfxr_set_error_writer"));
-
         {
             propagate_error_writer_t propagate_error_writer_to_hostfxr(set_error_writer_fn);
 
             rc = main_fn_v2(argc, argv, host_path_cstr, dotnet_root_cstr, app_path_cstr);
+
+            if (rc == StatusCode::FrameworkMissingFailure && !set_error_writer_fn)
```

Comment:
Should this just be for apphost GUI? In other cases (non-apphost or non-GUI apphost), the errors from hostfxr should already be traced to the terminal such that they are visible to the user.

Reply:
Great catch. Technically this should be done whenever the error writer for the apphost is redirected (which happens to be only for GUI apps, but maybe in the future there will be other cases). It is also easier to detect/explain than checking the GUI bit here (or somehow propagating that information).
But in that case it has to - as in if we don't redirect error writer we must not report the error second time as it will produce very confusing output.